src/pybind/mgr/mgr_util.py

```diff
@@ -73,6 +73,22 @@ def run(self):
             logger.error("task exception: %s", e)
             raise
 
+@contextlib.contextmanager
+def lock_timeout_log(lock: Lock, timeout: int = 5) -> Iterator[bool]:
+    start = time.time()
+    warned = False
+    while True:
+        logger.debug("locking {} with {} timeout".format(lock, timeout))
+        locked = lock.acquire(timeout=timeout)
+        if locked:
+            logger.debug("locked {}".format(lock))
+            yield locked
+            return lock.release()
```

Comment:
nit, since `lock.release()` always returns `None`, I'd suggest just put
```python
WARN_AFTER = 30
locked = False
start = time.time()
while True:
  logger.debug("locking {} with {} timeout".format(lock, timeout))
  locked = lock.acquire(timeout=timeout)
  if locked:
    logger.debug("locked {}".format(lock))
    break
  if time.time() - start > WARN_AFTER:
    logger.info("possible deadlock acquiring {}".format(lock))
    # might want to give up or raise an exception?
    break
yield locked
if locked:
  lock.release()
```